gui/albula/interface.py

```diff
@@ -0,0 +1,76 @@
+import subprocess
+import os
+
+
+class AlbulaInterface:
+    _instance = None  # All these variables are used to create a singleton
+    _init_args = None
+    _process = None
+
+    def __new__(cls, *args, **kwargs):
+        if cls._instance is None:
+            cls._instance = super(AlbulaInterface, cls).__new__(cls)
+        return cls._instance
+
+    def __init__(self, *args, **kwargs):
+        
+        if self._init_args is None:
+            self._init_args = (args, kwargs)
+            print("Opening Albula")
+            self.use_separate_process = False
+            if "python_path" in kwargs:
+                self.use_separate_process = True
+
+            self.open(args, kwargs)
+
+    def _call(self, code):
+        if self._process:
+            print(code, file=self._process.stdin, flush=True)
```

Comment:
So this looks like it's passing the commands into the controller process. This is a bit hard to find, can you document this better?